### PR TITLE
Update lambda guide for request/response transformations

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

Update the guide for AWS lambda to include a section for `wrapAsApiGateway`, as well as the `requestTransformation` and `responseTransformation` features. The goal of this update is to document each way to manipulate a request to or response from a lambda using this plugin, including which are available in OSS and EE, and what benefit the EE features (`wrapAsApiGateway`/`unwrapAsApiGateway`) provide over the OSS features (`requestTransformation`/`responseTransformation`).

# Context

The similarities between these features led to confusion when users were attempting to configure Gloo Edge with lambdas

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md [hunk 1]
```diff
@@ -123,9 +123,41 @@ Create Gloo Edge `Upstream` and `VirtualService` resources to route requests to
...
+                 responseTransformation: true
+             upstream:
+               name: aws-upstream
+               namespace: gloo-system
+   {{< / highlight >}}
+
+
+### Wrap or unwrap responses as an<!-- AWS ALB or--> AWS API Gateway
+
+Enterprise customers may use Gloo Edge in place of an AWS ALB or AWS API Gateway. To do this, you can additionally configure the `unwrapAsAlb` setting or the `unwrapAsApiGateway` setting (Gloo Edge Enterprise only, version 1.12.0 or later) in the [AWS `destinationSpec`]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/aws/aws.proto.sk/" %}}) of the route to your Lambda upstream. These settings allow Gloo Edge to manipulate a response from an upstream Lambda in the same way as an AWS ALB or AWS API Gateway.
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md:160]:
```suggestion
In Gloo Edge Enterprise, you can use Gloo Edge in place of an AWS ALB or AWS API Gateway. To do this, configure the `unwrapAsAlb` setting or the `unwrapAsApiGateway` setting (Gloo Edge Enterprise only, version 1.12.0 or later) in the [AWS `destinationSpec`]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/aws/aws.proto.sk/" %}}) of the route to your Lambda upstream. These settings allow Gloo Edge to manipulate a response from an upstream Lambda in the same way as an AWS ALB or AWS API Gateway.
```

File: docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md [hunk 2]
```diff
@@ -123,9 +123,41 @@ Create Gloo Edge `Upstream` and `VirtualService` resources to route requests to
...
-## Unwrap responses as an<!-- AWS ALB or--> AWS API Gateway
 
-To use Gloo Edge in place of your AWS ALB or AWS API Gateway, you can additionally configure the `unwrapAsAlb` setting or the `unwrapAsApiGateway` setting (Gloo Edge Enterprise only, version 1.12.0 or later) in the [AWS `destinationSpec`]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/aws/aws.proto.sk/" %}}) of the route to your Lambda upstream. These settings allow Gloo Edge to manipulate a response from an upstream Lambda in the same way as an AWS ALB or AWS API Gateway.
+## Transform requests and/or responses
+
+### Basic transformations
+
+Common use cases for this plugin could benefit from the option to transform requests and/or responses to align more closely with the AWS ecosystem.
+
+The request transformation, if enabled, will set inject the headers of the request into the body. The resulting body will be a JSON object containing two keys: 'headers' containing the injected headers, and 'body' containing the original body.
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md:133]:
```suggestion
The request transformation injects the headers of the request into the body. The resulting body will be a JSON object containing two keys: 'headers', which contains the injected headers, and 'body', which contains the original body.
```

File: docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md [hunk 3]
```diff
@@ -123,9 +123,41 @@ Create Gloo Edge `Upstream` and `VirtualService` resources to route requests to
...
+
+The response transformation, if enabled, will extract from the JSON response body returned from the upstream the value of the key 'body', and replace the _entire_ response body with this value. Additionally, the response transformation will overwrite the 'Content-Type' header to `text/html`. It should be noted that these transformations are incompatible with the features `wrapAsApiGateway`, `unwrapAsApiGateway`, and `unwrapAsAlb`.
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md:135]:
```suggestion
The response transformation extracts the value of the 'body' key from the upstream response's JSON body, and replaces the _entire_ response body with this value. Additionally, the response transformation overwrites the value of the 'Content-Type' header to `text/html`.

Note that these request and response transformations are incompatible with the features `wrapAsApiGateway`, `unwrapAsApiGateway`, and `unwrapAsAlb`.
```

File: docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md [hunk 4]
```diff
@@ -123,9 +123,41 @@ Create Gloo Edge `Upstream` and `VirtualService` resources to route requests to
...
 At this point, Gloo Edge is routing directly to the `echo` Lambda function. To configure Gloo Edge to also unwrap the JSON response from the function in the same way as an AWS ALB or AWS API Gateway, continue to the next section.
 
-## Unwrap responses as an<!-- AWS ALB or--> AWS API Gateway
 
-To use Gloo Edge in place of your AWS ALB or AWS API Gateway, you can additionally configure the `unwrapAsAlb` setting or the `unwrapAsApiGateway` setting (Gloo Edge Enterprise only, version 1.12.0 or later) in the [AWS `destinationSpec`]({{% versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options/aws/aws.proto.sk/" %}}) of the route to your Lambda upstream. These settings allow Gloo Edge to manipulate a response from an upstream Lambda in the same way as an AWS ALB or AWS API Gateway.
+## Transform requests and/or responses
+
+### Basic transformations
+
+Common use cases for this plugin could benefit from the option to transform requests and/or responses to align more closely with the AWS ecosystem.
```
Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md:131]:
```suggestion
When you use the AWS Lambda plug-in in Gloo Edge, you might want to transform requests and/or responses to align more closely with the AWS ecosystem.
```

File: docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md [hunk 5]
```diff
@@ -123,9 +123,43 @@ Create Gloo Edge `Upstream` and `VirtualService` resources to route requests to
...
+
+When you use the AWS Lambda plug-in in Gloo Edge, you might want to transform requests and/or responses to align more closely with the AWS ecosystem.
+
+The request transformation injects the headers of the request into the body. The resulting body will be a JSON object containing two keys: 'headers', which contains the injected headers, and 'body', which contains the original body.
+
+The response transformation extracts the value of the 'body' key from the upstream response's JSON body, and replaces the _entire_ response body with this value. Additionally, the response transformation overwrites the value of the 'Content-Type' header to `text/html`.
+
+Note that these request and response transformations are incompatible with the features `wrapAsApiGateway`, `unwrapAsApiGateway`, and `unwrapAsAlb`.
+
+**Before you begin**: [Install Gloo Edge Enterprise version 1.12.0 or later in a Kubernetes cluster]({{% versioned_link_path fromRoot="/installation/gateway/kubernetes/" %}}) or [upgrade your existing Enterprise installation to version 1.12.0 or later]({{% versioned_link_path fromRoot="/operations/upgrading/upgrade_steps/" %}}).
```
Comment (jbohanon) [inline docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md:139]: This does not require Gloo Edge Enterprise

Comment (Rachael-Graham) [inline docs/content/guides/traffic_management/destination_types/aws_lambda/_index.md:139]:
```suggestion
**Before you begin**: [Install Gloo Edge version 1.12.0 or later in a Kubernetes cluster]({{% versioned_link_path fromRoot="/installation/gateway/kubernetes/" %}}) or [upgrade your existing installation to version 1.12.0 or later]({{% versioned_link_path fromRoot="/operations/upgrading/upgrade_steps/" %}}).
```

**Review by Rachael-Graham — APPROVED** [review]

**Review by Nadine2016 — APPROVED** [review]

**Review by Rachael-Graham — APPROVED** [review]

**Review by Nadine2016 — APPROVED** [review]

**Pull request merged**